Source: bugs.gnome.org:567958
Reason: Fix theming issues
Upstream: Yes
--- 
--- gnome-session-2.25.91/gnome-session/gs-idle-monitor.c	2009-02-16 23:44:27.000000000 -0500
+++ gnome-session-2.25.91/gnome-session/gs-idle-monitor.c	2009-02-26 15:09:36.247212669 -0500
@@ -273,13 +273,6 @@
                 return FALSE;
         }
 
-        /* select for sync events */
-        gdk_error_trap_push ();
-        XSelectInput (GDK_DISPLAY (), GDK_ROOT_WINDOW (), XSyncAlarmNotifyMask);
-        if (gdk_error_trap_pop ()) {
-                g_warning ("XSelectInput failed");
-        }
-
         gdk_window_add_filter (NULL, (GdkFilterFunc)xevent_filter, monitor);
 
         return TRUE;
@@ -440,13 +433,15 @@
                 | XSyncCAValueType
                 | XSyncCATestType
                 | XSyncCAValue
-                | XSyncCADelta;
+                | XSyncCADelta
+                | XSyncCAEvents;
 
         XSyncIntToValue (&delta, 0);
         attr.trigger.counter = monitor->priv->counter;
         attr.trigger.value_type = XSyncAbsolute;
         attr.trigger.wait_value = watch->interval;
         attr.delta = delta;
+        attr.events = TRUE;
 
         attr.trigger.test_type = XSyncPositiveTransition;
         if (watch->xalarm_positive != None) {
